### PR TITLE
[RCL-97] civis_ml plot and histogram

### PR DESCRIPTION
**Pull request by patr1ckm**
Some diagnostic ggplots for CivisML! When calling `plot(my_model)` you get the decile plot for classification, and y-yhat plot for regression. Calling `hist(my_model)` gets the distribution of out of sample scores. 

These were inspired from the plots in platform and in model_deckR (thanks Ola and Bill!).

By default, these use theme_classic but with civis colors. You can easily add ` + civis.deckR::theme_civis()` to the result.

@olatop, @harader, @zachlipp , @AdvikS, @keithing for opinions and vis. If this is not very well aligned with deliverables, let me know. Happy to hear opinions on facets and defaults (all classes? one class?), let me know what would be most helpful.

 @wlattner for CR

Classification plot:
image

Regression plot:
image

Histogram:
image

Details:
- `hist.civis_ml` plots the histogram of the out of sample scores
- `plot.civis_ml_classifier` produces the decile plot
- `plot.civis_ml_regressor` produces the y_yhat plot
- `get_metric` is added to utils for getting a given metric by name (or all metrics)
- ggplot2 is added as a dependency.

### CODE REVIEW

File: R/civis_ml_plot.R [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+#' @export
+plot.civis_ml_regressor <- function(x, ...) {
+  pl <- get_metric(x, "y_yhat_plot")
+  if (is.null(pl)) stop("Plotting data not available.")
+
+  values <- data.frame(pl$values)
+  df <- utils::stack(values)
+  df$col_id <- as.numeric(df$ind)
+  df$row_id <- rep(1:nrow(values), ncol(values))
+  y_vals <- round(seq(from = pl$y_range[1], to = pl$y_range[2], by = pl$y_step))
```
Comment (wlattner) [inline R/civis_ml_plot.R]: Can you add a comment here describing the formatting of y_yhat_plot?

File: R/civis_ml_plot.R [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+  ggplot2::ggplot(df, ggplot2::aes_string(x = "x", y = "y", fill = "values")) +
+    ggplot2::geom_tile() +
+    ggplot2::scale_fill_gradient("Bin Size", low = "white", high = civisblue) +
+    ggplot2::geom_abline(intercept = 0, slope = 1, color = civisyellow) +
+    ggplot2::ylab("Predictions") + ggplot2::xlab("Actuals") +
+    ggplot2::theme_classic()
+}
+
+#' @export
+plot.civis_ml_classifier <- function(x, name =  NULL, ...) {
```
Comment (wlattner) [inline R/civis_ml_plot.R]: `name` here is the class name?

File: R/civis_ml_utils.R [hunk 3]
```diff
@@ -56,6 +56,26 @@ civis_ml_error <- function(civis_error_obj) {
             args = attr(civis_error_obj, "args"))
 }
 
+#' Get model metrics
+#' @param model Object from civis_ml
+#' @param name Name of the metric.
+#' @export
+#' @return The metric given by \code{metric_name}. If \code{NULL}, all metrics are returned.
+get_metric <- function(model, name = NULL) {
+  if (!is.null(name)) {
```
Comment (wlattner) [inline R/civis_ml_utils.R]: maybe check `is_civis_ml` here since it's exported

**Comment by wlattner** [comment]
This is great! Just a few minor code comments, I'll defer to others for the plots.

File: R/civis_ml_plot.R (hunk 2 above)
Comment (patr1ckm) [reply R/civis_ml_plot.R]: I'll document this better

File: R/civis_ml_plot.R [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+  y_vals <- round(seq(from = pl$y_range[1], to = pl$y_range[2], by = pl$y_step))
+  yhat_vals <- round(seq(from = pl$yhat_range[1], to = pl$yhat_range[2], by = pl$yhat_step))
+
+  df$x <- y_vals[df$row_id]
+  df$y <- yhat_vals[df$col_id]
+  ggplot2::ggplot(df, ggplot2::aes_string(x = "x", y = "y", fill = "values")) +
+    ggplot2::geom_tile() +
+    ggplot2::scale_fill_gradient("Bin Size", low = "white", high = civisblue) +
+    ggplot2::geom_abline(intercept = 0, slope = 1, color = civisyellow) +
+    ggplot2::ylab("Predictions") + ggplot2::xlab("Actuals") +
```
Comment (keithing) [inline R/civis_ml_plot.R]: Super minor point, but this chart has Proper Capitalization while the others are in ALL CAPS

**Pull request merged**